### DISCUSSION TITLE
Cookie com nível

### DISCUSSION THREAD

**Discussion by mobile-tramontina** [viniciussanchez/RESTRequest4Delphi#178] [General]
Olá galera

Estou quebrando a cabeça aqui para criar um cookie nivelado.

Exemplo:
```
  resp := TRequest.New.BaseURL('https://apis.averbeporto.com.br/websys/php/conn.php')
    .AddCookie('portal', 'ses=' + CookieText)
    .AddParam('comp', '5', pkREQUESTBODY)
    .AddParam('mod', 'Upload', pkREQUESTBODY)
    .AddParam('path', 'eguarda/php/', pkREQUESTBODY)
    .AddParam('dump', '1', pkREQUESTBODY)
    .AddFile('file', 'C:\nota_porto.xml')
    .Post;
```
Porém desse jeito está gerando conforme a imagem abaixo:
![RestRequest4D](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/73296658/14f23675-65d6-4bdf-97e3-bf9a732daba5)

E deveria gerar igual a imagem:
![preciso_gerar_assim](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/73296658/0ebf6201-67bd-4264-b35d-15e34dd0e3d1)

Seguindo a ideia de outras linguagens o exemplo abaixo deveria gerar em nível, porém tentei algumas formas sem sucesso.
.AddCookie('portal[ses]', CookieText)

Please HELP ME. Estou usando o Delphi 10.4.

**Comment by mobile-tramontina** [comment]
Olá galera

Depois de muitas tentativas, consegui achar uma solução para a situação apontada acima.

Precisei incluir um teste na unit **RESTRequest4D.Request.Client**
![Request client](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/73296658/e3a28ef2-5cfa-4716-88fa-c4b6b1a86fb4)

Com isso a requisição é criada conforme os requisitos necessários, porém o ideal é que essa opção pudesse ser apontada de forma opcional na function AddParam.

![sugestao_melhoria](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/73296658/2816ec72-5477-4726-bbc6-63d41bdb59d2)

Fica de sugestão caso acharem útil.

**Reply by viniciussanchez** [reply]
Implementado no commit: https://github.com/viniciussanchez/RESTRequest4Delphi/commit/5b0ede8e313555dd2992927a2e6511bda956b3ee

**Reply by Ptubbie** [reply]
Boa tarde, 
Vinicius estou tentando envio de dessa requisicao para o itau e nao aceita, quando coloco o pkREQUESTBODY tambem nao aceita, 
como fazer pra enviar??

![erroRest](https://github.com/user-attachments/assets/095ae42e-cb32-4c29-a93e-1c646ce4738d)
